packages/blocks/src/database-block/components/database-title.ts

```diff
@@ -0,0 +1,131 @@
+import { VEditor } from '@blocksuite/virgo';
+import { css, html } from 'lit';
+import { customElement, property, query } from 'lit/decorators.js';
+
+import { ShadowlessElement } from '../../__internal__/utils/lit.js';
+import { setupVirgoScroll } from '../../__internal__/utils/virgo.js';
+import { tooltipStyle } from '../../components/tooltip/tooltip.js';
+import type { DatabaseBlockModel } from '../database-model.js';
+
+@customElement('affine-database-title')
+export class DatabaseTitle extends ShadowlessElement {
+  static styles = css`
+    .affine-database-title {
+      flex: 1;
+      max-width: 300px;
+      min-width: 300px;
+      height: 30px;
+    }
+
+    .database-title {
+      flex: 1;
+      position: sticky;
+      width: 300px;
+      height: 30px;
+      font-size: 18px;
+      font-weight: 600;
+      line-height: 24px;
+      color: #424149;
+      font-family: inherit;
+      overflow: hidden;
+      cursor: text;
+    }
+
+    .database-title [data-virgo-text='true'] {
+      display: inline-block;
+      width: 300px;
+      max-width: 300px;
+      white-space: nowrap !important;
+      text-overflow: ellipsis;
+      overflow: hidden;
+    }
+
+    .database-title:focus {
+      outline: none;
+    }
+
+    .database-title:disabled {
+      background-color: transparent;
+    }
+
+    .database-title-empty::before {
+      content: 'Database';
+      color: var(--affine-placeholder-color);
+      position: absolute;
+      opacity: 0.5;
+    }
+
+    ${tooltipStyle}
+  `;
+
+  @property()
+  targetModel!: DatabaseBlockModel;
+
+  @property()
+  addRow!: (rowIndex?: number) => void;
+
+  @query('.database-title')
+  private _titleContainer!: HTMLDivElement;
+
+  private _vEditor!: VEditor;
+
+  firstUpdated() {
+    this._initTitleVEditor();
+  }
+
+  private _onShowTitleTooltip = () => {
+    // TODO: show tooltip according to title content(vEditor)
+  };
+
+  private _initTitleVEditor() {
+    this._vEditor = new VEditor(this.targetModel.title.yText);
+    setupVirgoScroll(this.targetModel.page, this._vEditor);
+    this._vEditor.mount(this._titleContainer);
+    this._vEditor.bindHandlers({
+      keydown: this._handleKeyDown,
+    });
+    this._vEditor.setReadonly(this.targetModel.page.readonly);
+
+    // for title placeholder
+    this.targetModel.title.yText.observe(() => {
+      this.requestUpdate();
+    });
+
+    // after the database structure is created
+    requestAnimationFrame(() => {
+      this._vEditor?.focusEnd();
+    });
```

Comment:
I believe we should not perform effects during the render process.

Reply:
> It should be moved after the operation of creating the databse.

This is more intuitive. I will change it.